lib/slayer/form.rb

```diff
@@ -1,17 +1,31 @@
+require 'virtus'
+require 'dry-validation'
+
 module Slayer
   class Form
-    extend ModelAttribute
-    include ActiveModel::Validations if defined?(Rails)
+    attr_reader :errors
+    include Virtus.model
 
-    # Internal: Form Class Methods
     class << self
-      def from_params(params)
-        raise NotImplementedError
-      end
+      attr_accessor :_schema
 
-      def from_model(model)
-        raise NotImplementedError
+      def validations(&block)
+        self._schema = Dry::Validation.Schema(&block)
       end
     end
+
+    def validate!
+      result  = self.class._schema.call(self.attributes)
+      @errors = result.errors
+    end
+
+    def valid?
```

Comment:
Similar to the thing above, it might be nice of `valid?` or `invalid?` could optionally take a schema (and directly return any results without affecting the state of the object).

This would let you have a "one-time" schema that you use for a Form object that has a different validation.